src/frontend/js/hooks/useContracts/index.tsx

```diff
@@ -22,8 +22,45 @@ const props: UseResourcesProps<Contract, ContractFilters, API['user']['contracts
   session: true,
   messages,
 };
+
 /**
  * Joanie Api hook to retrieve/update a contract owned by the authenticated user.
  */
-export const useContract = useResource(props);
-export const useContracts = useResources(props);
+export const useUserContract = useResource(props);
+export const useUserContracts = useResources(props);
+
+/**
+ * Joanie Api hook to retrieve/update a contracts related to a course.
+ */
+const organizationProps: UseResourcesProps<
+  Contract,
+  ContractFilters,
+  API['organizations']['contracts']
+> = {
+  ...props,
+  queryKey: ['organization_contracts'],
+  apiInterface: () => useJoanieApi().organizations.contracts,
+};
+
+export const useOrganizationContract = (
+  id: string,
+  filters: ContractFilters,
+  queryOptions?: QueryOptions<Contract>,
+) => {
+  return useResource(organizationProps)(id, filters, {
+    ...queryOptions,
+    enabled:
+      !!filters?.organization_id && (queryOptions?.enabled === undefined || queryOptions.enabled),
```

Comment:
If no id is provided, the query should be disabled.

```suggestion
     !!id && !!filters?.organization_id && (queryOptions?.enabled === undefined || queryOptions.enabled),
```